src/client/request.go

```diff
@@ -170,7 +169,7 @@ func (r *Request) invokeListeners() {
 		// Invoke next listener if present
 		listener := r.nextListener()
 		if listener != nil {
-			r.Response = listener.Invoke(r.Response)
```

Comment:
Tu nastaval niekedy problem, ze v pripade viacerych paralelnych requestov (a hlavne sub-requestov) nastavala `data-race` a k `r.Response` pristupovalo viacero vlakien naraz.

Riesenim je:
- 1. pridat zamok
- 2. alebo na tomto mieste do `response` nezapisovat

Pozrel som kod, a zvolil som variantu cislo 2. 
Povodne to bolo myslene tak, ze z `listener` je mozne modifokovat alebo aj uplne nahradit `listener`.
V praxi sa uplne nahradie response v tomto projekte nikde nepouiva, tak som to na vsetkych miestach zjednoducsil.